### PR TITLE
[21CENTURY-76] Retrieve correct information when requesting other user's grades

### PR DESCRIPTION
**Pull request by drtwisted**
**Description:**
Retrieve correct information when requesting other user's grades

**Youtrack:** 
https://youtrack.raccoongang.com/issue/F21C-76

### CODE REVIEW

File: lms/djangoapps/grades/api/views.py [hunk 1]
```diff
@@ -153,8 +155,20 @@ def get(self, request, course_id):
         if isinstance(course, Response):
             return course
 
+        if request.user.username == username:
+            graded_user = request.user
```
Comment (cmltaWt0) [inline lms/djangoapps/grades/api/views.py]: Why di we need the additional variable `graded_user `?

File: lms/djangoapps/grades/api/views.py [hunk 2]
```diff
@@ -153,8 +155,20 @@ def get(self, request, course_id):
         if isinstance(course, Response):
             return course
 
+        if request.user.username == username:
```
Comment (cmltaWt0) [inline lms/djangoapps/grades/api/views.py]: If it is better to use `not`?

```suggestion
        if request.user.username != username:
            user = User.objects.filter(username=username).first()
        if not user:
            return self.make_error_response(
                    status_code=status.HTTP_404_NOT_FOUND,
                    developer_message='The requested user does not exist.',
                    error_code='no_user'
                )
```

File: lms/djangoapps/grades/api/views.py (hunk 1 above)
Comment (drtwisted) [reply lms/djangoapps/grades/api/views.py]: @cmltaWt0 just to emphasize it's not necessarily the user that performs the request, but rather a user to get grades for.

File: lms/djangoapps/grades/api/views.py (hunk 2 above)
Comment (drtwisted) [reply lms/djangoapps/grades/api/views.py]: @cmltaWt0 Yes this looks cleaner!

File: lms/djangoapps/grades/api/views.py [hunk 3]
```diff
@@ -149,15 +151,26 @@ def get(self, request, course_id):
                 error_code='user_mismatch'
             )
 
-        course = self._get_course(course_id, user, 'load')
+        course = self._get_course(course_id, request.user, 'load')
         if isinstance(course, Response):
             return course
 
+        if request.user.username == username:
+            user = request.user
```
Comment (cmltaWt0) [inline lms/djangoapps/grades/api/views.py]: My second point was to overwrite the user only when needed - so only in second case.

File: lms/djangoapps/grades/api/views.py [hunk 4]
```diff
@@ -134,13 +136,13 @@ def get(self, request, course_id):
         username = request.GET.get('username')
         course_key = CourseKey.from_string(course_id)
 
-        is_staff_user = GlobalStaff().has_user(user) or CourseStaffRole(course_key).has_user(user)
+        is_staff_user = GlobalStaff().has_user(request.user) or CourseStaffRole(course_key).has_user(request.user)
```
Comment (cmltaWt0) [inline lms/djangoapps/grades/api/views.py]: Can't get the point in using `request.user` instead of `user`.

**Pull request merged**